diff --git a/vanetza/asn1/its/ConnectionManeuverAssist-addGrpC.c b/vanetza/asn1/its/ConnectionManeuverAssist-addGrpC.c
index a5ef3933..5609b8ee 100644
--- a/vanetza/asn1/its/ConnectionManeuverAssist-addGrpC.c
+++ b/vanetza/asn1/its/ConnectionManeuverAssist-addGrpC.c
@@ -6,6 +6,7 @@
  */
 
 #include "ConnectionManeuverAssist-addGrpC.h"
+#include "ItsStationPositionList.h"
 
 asn_TYPE_member_t asn_MBR_ConnectionManeuverAssist_addGrpC_1[] = {
 	{ ATF_POINTER, 1, offsetof(struct ConnectionManeuverAssist_addGrpC, itsStationPosition),
diff --git a/vanetza/asn1/its/ConnectionManeuverAssist-addGrpC.h b/vanetza/asn1/its/ConnectionManeuverAssist-addGrpC.h
index 2a3438c4..a9937684 100644
--- a/vanetza/asn1/its/ConnectionManeuverAssist-addGrpC.h
+++ b/vanetza/asn1/its/ConnectionManeuverAssist-addGrpC.h
@@ -43,7 +43,7 @@ extern asn_TYPE_member_t asn_MBR_ConnectionManeuverAssist_addGrpC_1[1];
 #endif
 
 /* Referred external types */
-#include "ItsStationPositionList.h"
+/* #include "ItsStationPositionList.h" */
 
 #endif	/* _ConnectionManeuverAssist_addGrpC_H_ */
 #include "asn_internal.h"
diff --git a/vanetza/asn1/its/MapData-addGrpC.c b/vanetza/asn1/its/MapData-addGrpC.c
index 18726629..05ac3a17 100644
--- a/vanetza/asn1/its/MapData-addGrpC.c
+++ b/vanetza/asn1/its/MapData-addGrpC.c
@@ -6,6 +6,7 @@
  */
 
 #include "MapData-addGrpC.h"
+#include "SignalHeadLocationList.h"
 
 asn_TYPE_member_t asn_MBR_MapData_addGrpC_1[] = {
 	{ ATF_POINTER, 1, offsetof(struct MapData_addGrpC, signalHeadLocations),
diff --git a/vanetza/asn1/its/MapData-addGrpC.h b/vanetza/asn1/its/MapData-addGrpC.h
index 4546458a..5c650c6d 100644
--- a/vanetza/asn1/its/MapData-addGrpC.h
+++ b/vanetza/asn1/its/MapData-addGrpC.h
@@ -43,7 +43,7 @@ extern asn_TYPE_member_t asn_MBR_MapData_addGrpC_1[1];
 #endif
 
 /* Referred external types */
-#include "SignalHeadLocationList.h"
+/* #include "SignalHeadLocationList.h" */
 
 #endif	/* _MapData_addGrpC_H_ */
 #include "asn_internal.h"
diff --git a/vanetza/asn1/its/NodeSetXY.c b/vanetza/asn1/its/NodeSetXY.c
index 5995decd..42f27803 100644
--- a/vanetza/asn1/its/NodeSetXY.c
+++ b/vanetza/asn1/its/NodeSetXY.c
@@ -6,6 +6,7 @@
  */
 
 #include "NodeSetXY.h"
+#include "NodeXY.h"
 
 #if !defined(ASN_DISABLE_OER_SUPPORT)
 static asn_oer_constraints_t asn_OER_type_NodeSetXY_constr_1 CC_NOTUSED = {
diff --git a/vanetza/asn1/its/NodeSetXY.h b/vanetza/asn1/its/NodeSetXY.h
index d10a2c1e..5c1b3c51 100644
--- a/vanetza/asn1/its/NodeSetXY.h
+++ b/vanetza/asn1/its/NodeSetXY.h
@@ -41,7 +41,7 @@ extern asn_per_constraints_t asn_PER_type_NodeSetXY_constr_1;
 #endif
 
 /* Referred external types */
-#include "NodeXY.h"
+/* #include "NodeXY.h" */
 
 #endif	/* _NodeSetXY_H_ */
 #include "asn_internal.h"
