diff --git a/vanetza/asn1/support/INTEGER_oer.c b/vanetza/asn1/support/INTEGER_oer.c
index 8e0db3244..40dd53f4c 100644
--- a/vanetza/asn1/support/INTEGER_oer.c
+++ b/vanetza/asn1/support/INTEGER_oer.c
@@ -49,7 +49,7 @@ INTEGER_decode_oer(const asn_codec_ctx_t *opt_codec_ctx,
         size -= consumed;
     }
 
-    if(req_bytes > size) {
+    if(req_bytes == 0 || req_bytes > size) {
         ASN__DECODE_STARVED;
     }
 
