diff --git a/vanetza/asn1/support/asn_system.h b/vanetza/asn1/support/asn_system.h
index 51859cdd..c97a9ad5 100644
--- a/vanetza/asn1/support/asn_system.h
+++ b/vanetza/asn1/support/asn_system.h
@@ -125,7 +125,7 @@ typedef	unsigned int	uint32_t;
 #endif /* __GNUC__ */
 #endif	/* MIN */
 
-#if __STDC_VERSION__ >= 199901L
+#if __STDC_VERSION__ >= 199901L || __cplusplus >= 201103L
 #ifndef SIZE_MAX
 #define SIZE_MAX   ((~((size_t)0)) >> 1)
 #endif
@@ -145,7 +145,7 @@ typedef	unsigned int	uint32_t;
 #define RSSIZE_MAX   ((ssize_t)(RSIZE_MAX >> 1))
 #endif
 
-#if __STDC_VERSION__ >= 199901L
+#if __STDC_VERSION__ >= 199901L || __cplusplus >= 201103L
 #define ASN_PRI_SIZE "zu"
 #define ASN_PRI_SSIZE "zd"
 #define ASN_PRIuMAX PRIuMAX
